extension/arquillian-drone-browserstack-extension/src/main/java/org/arquillian/drone/browserstack/extension/webdriver/BrowserStackDriverFactory.java

```diff
@@ -0,0 +1,130 @@
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.arquillian.drone.browserstack.extension.webdriver;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.logging.Logger;
+
+import org.arquillian.drone.browserstack.extension.local.BrowserStackLocalRunner;
+import org.jboss.arquillian.config.descriptor.api.ArquillianDescriptor;
+import org.jboss.arquillian.core.api.Instance;
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.drone.spi.Configurator;
+import org.jboss.arquillian.drone.spi.Destructor;
+import org.jboss.arquillian.drone.spi.DronePoint;
+import org.jboss.arquillian.drone.spi.Instantiator;
+import org.jboss.arquillian.drone.webdriver.configuration.WebDriverConfiguration;
+import org.jboss.arquillian.drone.webdriver.spi.BrowserCapabilities;
+import org.jboss.arquillian.drone.webdriver.spi.BrowserCapabilitiesRegistry;
+import org.openqa.selenium.Capabilities;
+
+import static org.arquillian.drone.browserstack.extension.webdriver.BrowserStackCapabilities.ACCESS_KEY;
+import static org.arquillian.drone.browserstack.extension.webdriver.BrowserStackCapabilities.BROWSERSTACK_LOCAL;
+import static org.arquillian.drone.browserstack.extension.webdriver.BrowserStackCapabilities.BROWSERSTACK_LOCAL_ARGS;
+import static org.arquillian.drone.browserstack.extension.webdriver.BrowserStackCapabilities.BROWSERSTACK_LOCAL_BINARY;
+import static org.arquillian.drone.browserstack.extension.webdriver.BrowserStackCapabilities.BROWSERSTACK_LOCAL_MANAGED;
+import static org.arquillian.drone.browserstack.extension.webdriver.BrowserStackCapabilities.READABLE_NAME;
+import static org.arquillian.drone.browserstack.extension.webdriver.BrowserStackCapabilities.URL;
+import static org.arquillian.drone.browserstack.extension.webdriver.BrowserStackCapabilities.USERNAME;
+
+/**
+ * Factory which combines {@link org.jboss.arquillian.drone.spi.Configurator},
+ * {@link org.jboss.arquillian.drone.spi.Instantiator} and {@link org.jboss.arquillian.drone.spi.Destructor} for
+ * {@link BrowserStackDriver}.
+ *
+ * @author <a href="mailto:mjobanek@redhat.com">Matous Jobanek</a>
+ */
+public class BrowserStackDriverFactory implements
+    Configurator<BrowserStackDriver, WebDriverConfiguration>,
+    Instantiator<BrowserStackDriver, WebDriverConfiguration>,
+    Destructor<BrowserStackDriver> {
+
+    private static final Logger log = Logger.getLogger(BrowserStackDriverFactory.class.getName());
+
+    @Inject
+    protected Instance<BrowserCapabilitiesRegistry> registryInstance;
```

Comment:
are you sure it has to be protected? i would say private is the keyword you want.